diff --git a/VSSurvivalMod/CollectibleBehavior/BehaviorWearable.cs b/VSSurvivalMod/CollectibleBehavior/BehaviorWearable.cs
index c08153f..b156542 100644
--- a/VSSurvivalMod/CollectibleBehavior/BehaviorWearable.cs
+++ b/VSSurvivalMod/CollectibleBehavior/BehaviorWearable.cs
@@ -8,12 +8,10 @@ using Vintagestory.API.Common.Entities;
 using Vintagestory.API.Config;
 using Vintagestory.API.Datastructures;
 using Vintagestory.API.MathTools;
 using Vintagestory.API.Util;
 
-#nullable disable
-
 namespace Vintagestory.GameContent;
 
 public class CollectibleBehaviorWearable : CollectibleBehaviorWearableAttachment, IWearable, IWearableStatsSupplier
 {
     private StatModifiers? StatModifers;
