diff --git a/VintagestoryLib/VintagestoryLib.csproj b/VintagestoryLib/VintagestoryLib.csproj
index 6e31dff..7f7d477 100644
--- a/VintagestoryLib/VintagestoryLib.csproj
+++ b/VintagestoryLib/VintagestoryLib.csproj
@@ -7,36 +7,34 @@
   <PropertyGroup>
     <LangVersion>latest</LangVersion>
     <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
     <CheckForOverflowUnderflow>False</CheckForOverflowUnderflow>
   </PropertyGroup>
-  <PropertyGroup />
-  <ItemGroup />
   <ItemGroup>
-    <Reference Include="VintagestoryAPI">
-      <HintPath>../../.vanilla/vintagestory/VintagestoryAPI.dll</HintPath>
-    </Reference>
-    <Reference Include="cairo-sharp" />
-    <Reference Include="protobuf-net" />
-    <Reference Include="Newtonsoft.Json" />
-    <Reference Include="CommandLine" />
-    <Reference Include="SkiaSharp" />
-    <Reference Include="Open.Nat" />
-    <Reference Include="Mono.Nat" />
-    <Reference Include="Mono.Cecil" />
-    <Reference Include="Microsoft.CodeAnalysis" />
-    <Reference Include="Microsoft.CodeAnalysis.CSharp" />
-    <Reference Include="ICSharpCode.SharpZipLib" />
-    <Reference Include="Microsoft.Data.Sqlite" />
-    <Reference Include="0Harmony" />
-    <Reference Include="OpenTK.Windowing.Desktop" />
-    <Reference Include="OpenTK.Audio.OpenAL" />
-    <Reference Include="OpenTK.Mathematics" />
-    <Reference Include="OpenTK.Windowing.Common" />
-    <Reference Include="OpenTK.Windowing.GraphicsLibraryFramework" />
-    <Reference Include="DnsClient" />
-    <Reference Include="OpenTK.Graphics" />
-    <Reference Include="csvorbis" />
-    <Reference Include="csogg" />
-    <Reference Include="xplatforminterface" />
+    <ProjectReference Include="..\..\VintagestoryApi\VintagestoryAPI.csproj" />
   </ItemGroup>
-</Project>
\ No newline at end of file
+  <ItemGroup>
+    <Reference Include="cairo-sharp"><HintPath>..\..\.vanilla\vintagestory\Lib\cairo-sharp.dll</HintPath></Reference>
+    <Reference Include="protobuf-net"><HintPath>..\..\.vanilla\vintagestory\Lib\protobuf-net.dll</HintPath></Reference>
+    <Reference Include="Newtonsoft.Json"><HintPath>..\..\.vanilla\vintagestory\Lib\Newtonsoft.Json.dll</HintPath></Reference>
+    <Reference Include="CommandLine"><HintPath>..\..\.vanilla\vintagestory\Lib\CommandLine.dll</HintPath></Reference>
+    <Reference Include="SkiaSharp"><HintPath>..\..\.vanilla\vintagestory\Lib\SkiaSharp.dll</HintPath></Reference>
+    <Reference Include="Open.Nat"><HintPath>..\..\.vanilla\vintagestory\Lib\Open.Nat.dll</HintPath></Reference>
+    <Reference Include="Mono.Nat"><HintPath>..\..\.vanilla\vintagestory\Lib\Mono.Nat.dll</HintPath></Reference>
+    <Reference Include="Mono.Cecil"><HintPath>..\..\.vanilla\vintagestory\Lib\Mono.Cecil.dll</HintPath></Reference>
+    <Reference Include="Microsoft.CodeAnalysis"><HintPath>..\..\.vanilla\vintagestory\Lib\Microsoft.CodeAnalysis.dll</HintPath></Reference>
+    <Reference Include="Microsoft.CodeAnalysis.CSharp"><HintPath>..\..\.vanilla\vintagestory\Lib\Microsoft.CodeAnalysis.CSharp.dll</HintPath></Reference>
+    <Reference Include="ICSharpCode.SharpZipLib"><HintPath>..\..\.vanilla\vintagestory\Lib\ICSharpCode.SharpZipLib.dll</HintPath></Reference>
+    <Reference Include="Microsoft.Data.Sqlite"><HintPath>..\..\.vanilla\vintagestory\Lib\Microsoft.Data.Sqlite.dll</HintPath></Reference>
+    <Reference Include="0Harmony"><HintPath>..\..\.vanilla\vintagestory\Lib\0Harmony.dll</HintPath></Reference>
+    <Reference Include="OpenTK.Windowing.Desktop"><HintPath>..\..\.vanilla\vintagestory\Lib\OpenTK.Windowing.Desktop.dll</HintPath></Reference>
+    <Reference Include="OpenTK.Audio.OpenAL"><HintPath>..\..\.vanilla\vintagestory\Lib\OpenTK.Audio.OpenAL.dll</HintPath></Reference>
+    <Reference Include="OpenTK.Mathematics"><HintPath>..\..\.vanilla\vintagestory\Lib\OpenTK.Mathematics.dll</HintPath></Reference>
+    <Reference Include="OpenTK.Windowing.Common"><HintPath>..\..\.vanilla\vintagestory\Lib\OpenTK.Windowing.Common.dll</HintPath></Reference>
+    <Reference Include="OpenTK.Windowing.GraphicsLibraryFramework"><HintPath>..\..\.vanilla\vintagestory\Lib\OpenTK.Windowing.GraphicsLibraryFramework.dll</HintPath></Reference>
+    <Reference Include="DnsClient"><HintPath>..\..\.vanilla\vintagestory\Lib\DnsClient.dll</HintPath></Reference>
+    <Reference Include="OpenTK.Graphics"><HintPath>..\..\.vanilla\vintagestory\Lib\OpenTK.Graphics.dll</HintPath></Reference>
+    <Reference Include="csvorbis"><HintPath>..\..\.vanilla\vintagestory\Lib\csvorbis.dll</HintPath></Reference>
+    <Reference Include="csogg"><HintPath>..\..\.vanilla\vintagestory\Lib\csogg.dll</HintPath></Reference>
+    <Reference Include="xplatforminterface"><HintPath>..\..\.vanilla\vintagestory\Lib\xplatforminterface.dll</HintPath></Reference>
+  </ItemGroup>
+</Project>
