diff --git a/VintagestoryLib/Vintagestory.Client.NoObf/ChunkTesselatorManager.cs b/VintagestoryLib/Vintagestory.Client.NoObf/ChunkTesselatorManager.cs
index 5714490..309af53 100644
--- a/VintagestoryLib/Vintagestory.Client.NoObf/ChunkTesselatorManager.cs
+++ b/VintagestoryLib/Vintagestory.Client.NoObf/ChunkTesselatorManager.cs
@@ -1,6 +1,8 @@
 using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
 using Vintagestory.API.Client;
 using Vintagestory.API.Datastructures;
 using Vintagestory.API.MathTools;
 using Vintagestory.Common.Database;
 
@@ -8,15 +10,15 @@ namespace Vintagestory.Client.NoObf;
 
 public class ChunkTesselatorManager : ClientSystem
 {
 	internal int chunksize;
 
-	private object tessChunksQueueLock = new object();
+	private readonly Lock tessChunksQueueLock = new();
 
 	private SortableQueue<TesselatedChunk> tessChunksQueue = new SortableQueue<TesselatedChunk>();
 
-	private object tessChunksQueuePriorityLock = new object();
+	private readonly Lock tessChunksQueuePriorityLock = new();
 
 	private Queue<TesselatedChunk> tessChunksQueuePriority = new Queue<TesselatedChunk>();
 
 	private Vec3i chunkPos = new Vec3i();
 
