diff --git a/VintagestoryLib/Vintagestory.Client.NoObf/ParticlePoolQuads.cs b/VintagestoryLib/Vintagestory.Client.NoObf/ParticlePoolQuads.cs
index 4db59d6..296842c 100644
--- a/VintagestoryLib/Vintagestory.Client.NoObf/ParticlePoolQuads.cs
+++ b/VintagestoryLib/Vintagestory.Client.NoObf/ParticlePoolQuads.cs
@@ -1,6 +1,8 @@
 using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
 using Vintagestory.API.Client;
 using Vintagestory.API.Common;
 using Vintagestory.API.Config;
 using Vintagestory.API.MathTools;
 using Vintagestory.Common;
@@ -25,11 +27,11 @@ public class ParticlePoolQuads : IParticlePool
 
 	private int writePosition = 1;
 
 	private int readPosition;
 
-	private object advanceCountLock = new object();
+	private readonly Lock advanceCountLock = new();
 
 	private int advanceCount;
 
 	protected int poolSize;
 
